mersenne-31/src/poseidon2.rs

```diff
@@ -173,34 +162,24 @@ mod tests {
     // See: https://github.com/0xPolygonZero/hash-constants for the sage code used to create all these tests.
 
     // Our Poseidon2 Implementation for Mersenne31
-    fn poseidon2_mersenne31<const WIDTH: usize, const D: u64>(input: &mut [F; WIDTH])
+    fn poseidon2_mersenne31<const WIDTH: usize>(input: &mut [F; WIDTH])
     where
-        Poseidon2ExternalLayerMersenne31: ExternalLayer<Mersenne31, WIDTH, D>,
+        Poseidon2ExternalLayerMersenne31<WIDTH>: ExternalLayer<Mersenne31, WIDTH, 5>,
         Poseidon2InternalLayerMersenne31: InternalLayer<
             Mersenne31,
             WIDTH,
-            D,
-            InternalState = <Poseidon2ExternalLayerMersenne31 as ExternalLayer<
+            5,
```

Comment:
This magic constant should be given a name, like `SBOX_DEGREE` or similar (same goes for all other occurrences of `5` here).

In fact, given that this value is no longer a parameter that can be set by the caller, maybe it shouldn't be specified here at all, but lower down?